src/lib.rs

```diff
@@ -701,14 +701,18 @@ impl<R: gimli::Reader> ResUnit<R> {
             String::new()
         };
 
-        if let Some(directory) = file.directory(header) {
-            path_push(
-                &mut path,
-                sections
-                    .attr_string(&self.dw_unit, directory)?
-                    .to_string_lossy()?
-                    .as_ref(),
-            );
+        // In DWARF version <= 4, the index 0 is defined to correspond to the
+        // compilation unit directory.
+        if !(header.version() <= 4 && file.directory_index() == 0) {
```

Comment:
DWARF 5 says: "The first entry is the current directory of the compilation. Each additional path entry is either a full path name or is relative to the current directory of the compilation. The line number program assigns a number (index) to each of the directory entries in order, beginning with 0."

So I don't think we should restrict this to only DWARF version <= 4.

Reply:
Interesting. Well the `LineProgramHeader` uses `<= 4` as a cutoff point, both for directories and file names further down below:

https://github.com/gimli-rs/gimli/blob/fdd7ee4ed4c58bee9b79ffdc895488b40a87c899/src/read/line.rs#L1168-L1190

Should I update all of that and rather continue with https://github.com/gimli-rs/gimli/pull/604 first?

Reply:
The change in version 5 (and the reason for <=4 as a cutoff in that link) is that previously the compilation directory entry was not included in `.debug_line`, but a directory index of 0 still means the same for all versions.

There is a separate issue where earlier in this function we probably should be using `header.directory(0)` instead of `self.dw_unit.comp_dir`, but in theory they should be equivalent.

Reply:
(So the code you linked to in `line.rs` doesn't need fixing, it is correct; all it is doing is handling the case where the compilation directory is not present in `self.include_directories` so it has to use `self.comp_dir` instead.)

Reply:
Thanks for the explanation!